### PR TITLE
Sci 1024 vcf pca

### PR DESCRIPTION
**Pull request by Arkarachai**
#dxapp.json
- [x] Upgrade to 14.04
- [x] Add help and extension to input spec.
- [x] In `details` field specify: `whatsNew`and `citation` (leave blank for citation.
- [x] In `details` consider specifying `upstreamAuthor` and `upstreamVersion` fields when applicable.
- [x] Use `regionalOptions` 
- [x] Set applicable timeout (Be relax) and restart policy in [`runSpec`](https://wiki.dnanexus.com/API-Specification-v1.0.0/IO-and-Run-Specifications).
- [x] Add subjob restartability when applicable.
- [x] `version` field, set proper [semantic versioning](https://semver.org/).
- [x] Setting `"openSource": true`. 
- [x] Run [dxapp linter](https://github.com/dnanexus/RoughER/blob/master/scripts/dxapp_lint.py) [[permalink](https://github.com/dnanexus/RoughER/blob/ecb3ea3f97870c020f93bf4723afc9f40a310e33/scripts/dxapp_lint.py)] on `dxapp.json`.
- [x] Use AssetDepend instead of execDepend
- [x] Add default instance type that works for a general use case.
- [x] add new input if applicable

#src/code.sh src/code.py
- [x] GATK tarball should be removed from the resource project (or make a new resource project), and the applet should require GATK tarball as a required input. This way, the users are enforced to provide their own tarball. This would avoid trouble from non-academic entities without GATK license to use your tarball.

# Readme.md
- [x] Improve Readme file to contain applet function, use case, input/output description, underline execution if it’s a wrapper.
- [ ] Read documentation for sanity check. (Unpublished app URL will be: https://platform.dnanexus.com/app/app-xxxx)

# Test
- [] Basic testing using given input in Box or expected input
- [ ] Make a Pull Request to `master` in https://github.com/RitchieLab/dnanexus

# Production
- [ ] Co-test with Ritchie group
- [ ] Help Ritchie group to publish (Give dx command)
- [ ] Verify the versions that appear on the platform are the intended versions
- [ ] Set up demo workflow in public project.

### CODE REVIEW

**Review by nainathangaraj — APPROVED** [review]
Naming convention and the help field in the dxapp.json need to be updated, and everything else is good :)

**Comment by Arkarachai** [comment]
@slzarate Could you check the README of this applet? Thanks.

File: sequencing/vcf_pca/Readme.md [hunk 1]
```diff
@@ -1,21 +1,18 @@
...
 # Calculate PCs from VCF (DNAnexus Platform App)
 
-Uses PLINK 1.9 and eigenstrat 6.0 to calculate principal components from VCF or PLINK bed/bim/fam files.
+## What does this applet do?
 
-This is the source code for an app that runs on the DNAnexus Platform.
-For more information about how to run or modify it, see
-https://wiki.dnanexus.com/.
-<!-- /dx-header -->
+This app uses PLINK 1.9 and eigenstrat 6.0 to calculate principal components from VCF (`*.vcf.gz`) or PLINK files(`*.bed`, `*.bim`, `*.fam`). The app can process both type of data at the same time.
```
Comment (slzarate) [inline sequencing/vcf_pca/Readme.md]: "files(`*.bed`, `*.bim`, `*.fam`)" --> "files (`*.bed`, `*.bim`, `*.fam`)"

"both type of data" --> "both types of data"

File: sequencing/vcf_pca/Readme.md [hunk 2]
```diff
@@ -1,21 +1,18 @@
...
 
+This app takes the following inputs:
 
-<!--
-TODO: This app directory was automatically generated by dx-app-wizard;
-please edit this Readme.md file to include essential documentation about
-your app that would be helpful to users. (Also see the
-Readme.developer.md.) Once you're done, you can remove these TODO
-comments.
+For VCF data, this app requires equal number of VCF (variant calling format) files(`*.vcf.gz`) and their associated index files (`*.vcf.gz.tbi`). VCF is the standard format for variant calling.
```
Comment (slzarate) [inline sequencing/vcf_pca/Readme.md]: Change to "For VCF data, this app requires an equal number of VCF (variant calling format) files (`*.vcf.gz`) and their associated index files (`*.vcf.gz.tbi`). VCF is the standard format for variant calling."

Also, do we have to explain what a VCF file is?

File: sequencing/vcf_pca/Readme.md [hunk 3]
```diff
@@ -1,21 +1,18 @@
...
 
-<!--
-TODO: This app directory was automatically generated by dx-app-wizard;
-please edit this Readme.md file to include essential documentation about
-your app that would be helpful to users. (Also see the
-Readme.developer.md.) Once you're done, you can remove these TODO
-comments.
+For VCF data, this app requires equal number of VCF (variant calling format) files(`*.vcf.gz`) and their associated index files (`*.vcf.gz.tbi`). VCF is the standard format for variant calling. 
+
+For PLINK, this app requires equal number of binary genotype files (`*.bed`), extend mapping files (`*.bim`), and FAM files (`*.fam`) -- first 6 columns of pedigree. PLINK is a standard format for whole genome association analysis of genotype/phenotype data.
```
Comment (slzarate) [inline sequencing/vcf_pca/Readme.md]: Change to "For PLINK, this app requires an equal number of binary genotype files (`*.bed`), extend mapping files (`*.bim`), and FAM files (`*.fam`) -- first 6 columns of pedigree. PLINK is a standard format for whole-genome association analysis of genotype/phenotype data."

Do we have to explain the PLINK format? Also, what does "first 6 columns of pedigree" mean?

File: sequencing/vcf_pca/dxapp.json [hunk 4]
```diff
@@ -1,9 +1,9 @@
 {
   "name": "vcf_pca",
   "title": "Calculate PCs from VCF",
-  "summary": "vcf_pca",
+  "summary": "Uses PLINK 1.9 and eigenstrat 6.0 to calculate principal components from VCF or PLINK bed/bim/fam files",
```
Comment (slzarate) [inline sequencing/vcf_pca/dxapp.json:4]: Shouldn't there be a period at the end of this sentence? Run this through the dxapp.json linter

File: sequencing/vcf_pca/dxapp.json [hunk 5]
```diff
@@ -91,28 +92,28 @@
...
-      "label" : "Calculate projection using super-populations (EUR, etc.)",
+      "label" : "Calculate projection using super-populations (EUR, etc.)?",
       "class" : "boolean",
       "optional" : true,
       "default" : false
     },
     {
       "name": "fast_pca",
-      "label": "Use Fast Eigenvalue Approx. (Non fast may not work)",
+      "label": "Use Fast Eigenvalue Approx? (Non fast may not work)",
```
Comment (slzarate) [inline sequencing/vcf_pca/dxapp.json]: "Non fast may not work" doesn't really make sense. Instead, maybe clarify with "Use Fast Eigenvalue Approx? (If false, app may fail.)" or something like that

File: sequencing/vcf_pca/dxapp.json [hunk 6]
```diff
@@ -161,6 +162,7 @@
       "name": "twstats_out",
       "label": "TW Statistics",
       "class": "file",
+      "help": "Tracy-Widom statistics to evaluate the statistical significance of each principal component identified by pca",
```
Comment (slzarate) [inline sequencing/vcf_pca/dxapp.json]: A file containing Tracy-Widom statistics evaluating the statistical significance of each principal component identified by PCA

**Review by slzarate — COMMENTED** [review]
Run these through Grammarly and dxapp.json

File: sequencing/vcf_pca/Readme.md (hunk 3 above)
Comment (Arkarachai) [reply sequencing/vcf_pca/Readme.md]: I think PLINK is a bit more complicated than VCF, so I wish to explain a bit.
"first 6 columns of pedigree" is the definition of fam file. I think I didn't say it well. It should be "first 6 columns of the pedigree file". Is that make sense?

File: sequencing/vcf_pca/Readme.md [hunk 7]
```diff
@@ -1,21 +1,19 @@
-<!-- dx-header -->
 # Calculate PCs from VCF (DNAnexus Platform App)
 
-Uses PLINK 1.9 and eigenstrat 6.0 to calculate principal components from VCF or PLINK bed/bim/fam files.
+## What does this applet do?
```
Comment (yihchii) [inline sequencing/vcf_pca/Readme.md]: "What does this applet do?" --> "What does this app do?"

File: sequencing/vcf_pca/Readme.md [hunk 8]
```diff
@@ -1,21 +1,19 @@
...
-please edit this Readme.md file to include essential documentation about
-your app that would be helpful to users. (Also see the
-Readme.developer.md.) Once you're done, you can remove these TODO
-comments.
+For VCF data, this app requires an equal number of VCF (variant calling format) files (`*.vcf.gz`) and their associated index files (`*.vcf.gz.tbi`)
+
+For PLINK, this app requires an equal number of binary genotype files (`*.bed`), extend mapping files (`*.bim`), and FAM files (`*.fam`) -- first 6 columns of the pedigree file. PLINK is a standard format for whole-genome association analysis of genotype/phenotype data.
+
+
+## What does this applet output?
```
Comment (yihchii) [inline sequencing/vcf_pca/Readme.md]: "What does this applet output?" --> "What does this app output?"

File: sequencing/vcf_pca/dxapp.json [hunk 9]
```diff
@@ -161,6 +162,7 @@
       "name": "twstats_out",
       "label": "TW Statistics",
       "class": "file",
+      "help": "A file containing Tracy-Widom statistics evaluating the statistical significance of each principal component identified by PCA",
```
Comment (yihchii) [inline sequencing/vcf_pca/dxapp.json]: Add period at the end of help. Run through dxapp linter.

File: sequencing/vcf_pca/dxapp.json [hunk 10]
```diff
@@ -171,16 +173,26 @@
...
+      "*": { "hours": 120 }
+    }
+  },
+  "regionalOptions": {
+    "aws:us-east-1": {"systemRequirements": {"*": {"instanceType": "mem1_ssd1_x16"}},
+                    "assetDepends": [{"id": "record-FGpf7zQ0kfbZxj6q4ZzQ2BPG"}],
+                    "resources" : "project-FGqF5B00ByvPXG4G87yQy23G"},
+    "azure:westus": {"systemRequirements": {"*": {"instanceType": "azure:mem2_ssd1_x16"}},
+                    "assetDepends": [{"id": "record-FGpfZ1j95qzXV43bP5y5JKgV"}],
+                    "resources" : "project-FGvp3009BJq40pkZ683v3754"}
```
Comment (yihchii) [inline sequencing/vcf_pca/dxapp.json]: Please update record IDs to:
aws:us-east-1: record-FJ3zf8j08zVZFzqYG88Pbk7z
azure:westus: record-FJ4058j9Vf9ZBBVB2Q7b3VQz

File: sequencing/vcf_pca/dxapp.json [hunk 11]
```diff
@@ -171,16 +173,26 @@
...
   },
-  "resources" : "project-BYpFk1Q0pB0xzQY8ZxgJFv1V",
   "openSource" : true,
-  "developers" : ["user-tnperson", "user-auv13", "user-atf3"],
-  "authorizedUsers": ["org-geisinger_affiliate", "org-geisinger", "org-support", "user-codushlaine", "user-semanti.mukherjee", "user-alexander.li", "user-shannon.bruse", "user-jan.freudenberg", "user-crisvh"]
+  "details": {
+    "upstreamVersion": "PLINK 1.9 and eigenstrat 6.0",
+    "whatsNew": ""
```
Comment (yihchii) [inline sequencing/vcf_pca/dxapp.json]: "whatsNew": "* 1.0.0 First version of the Calculate PCs from VCF app."

**Review by yihchii — COMMENTED** [review]
LGTM! 
Added some comments. Please do update the recordIDs and do testing before merging.

**Comment by yihhwang** [comment]
@slzarate , Please help review Readme.md, thanks!

**Pull request merged**